code/modules/antagonists/heretic/magic/moon_parade.dm

```diff
@@ -67,29 +68,30 @@
 		visible_message(span_warning("The parade hits [victim] and a sudden wave of clarity comes over you!"))
 		return PROJECTILE_DELETE_WITHOUT_HITTING
 
+	return ..()
 
 /obj/projectile/moon_parade/on_hit(atom/target, blocked = 0, pierce_hit)
 	. = ..()
+	if(. == BULLET_ACT_BLOCK || !isliving(target))
+		return
+
 	var/mob/living/victim = target
 
-	RegisterSignal(victim, COMSIG_MOB_CLIENT_PRE_LIVING_MOVE, PROC_REF(moon_block_move), override=TRUE)
+	if(!(victim in mobs_hit))
+		RegisterSignal(victim, COMSIG_MOB_CLIENT_PRE_LIVING_MOVE, PROC_REF(moon_block_move))
+		RegisterSignal(victim, COMSIG_QDELETING, PROC_REF(clear_mob))
+		victim.AddComponent(/datum/component/leash, src, distance = 1)
+		victim.balloon_alert(victim, "you feel unable to move away from the parade!")
+		mobs_hit += victim
 
-	victim.AddComponent(/datum/component/leash, src, distance = 1)
-	victim.balloon_alert(victim,"you feel unable to move away from the parade!")
 	victim.add_mood_event("Moon Insanity", /datum/mood_event/moon_insanity)
-	victim.cause_hallucination(/datum/hallucination/delusion/preset/moon, "delusion/preset/moon hallucination caused by lunar parade")
-
-	//Lowers sanity
+	victim.cause_hallucination(/datum/hallucination/delusion/preset/moon, name)
 	victim.mob_mood.set_sanity(victim.mob_mood.sanity - 20)
 
-	// Uses weakref to prevent qdeleting them
-	mobs_hit |= WEAKREF(victim)
```

Comment:
I don't understand why you're removing weakrefs in favor of qdel signals. The code you have before seems much better to me.

Reply:
I dunno it just seemed easier since we were already registering another signal. (`COMSIG_MOB_CLIENT_PRE_LIVING_MOVE`)

Reply:
Ah I understand